tests/tags_test.go

```diff
@@ -1,16 +1,146 @@
 package tests
 
 import (
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"sync"
+
 	. "github.com/onsi/ginkgo"
-	// . "github.com/onsi/gomega"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
 )
 
 var _ = Describe("Tags", func() {
+
 	Context("with a deployed app", func() {
+		var testApp App
+		once := &sync.Once{}
+
+		BeforeEach(func() {
+			// use the "kubectl" executable in the search $PATH
+			if _, err := exec.LookPath("kubectl"); err != nil {
+				Skip("kubectl not found in search $PATH")
+			}
+
+			// Set up the Tags test app only once and assume the suite will clean up.
+			once.Do(func() {
+				os.Chdir("example-go")
+				appName := getRandAppName()
+				createApp(appName)
+				testApp = deployApp(appName)
+			})
+		})
+
+		It("can set and unset tags", func() {
+			// can list tags
+			sess, err := start("deis tags:list")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Tags", testApp.Name))
+			Eventually(sess).Should(Exit(0))
+
+			// set an invalid tag
+			sess, err = start("deis tags:set munkafolyamat=yeah")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).ShouldNot(Say("=== %s Tags", testApp.Name))
+			Eventually(sess).ShouldNot(Say(`munkafolyamat\s+yeah`, testApp.Name))
+			Eventually(sess.Err).Should(Say("400 Bad Request"))
+			Eventually(sess).Should(Exit(1))
+
+			// list tags
+			sess, err = start("deis tags:list")
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Tags", testApp.Name))
+			Eventually(sess).ShouldNot(Say(`munkafolyamat\s+yeah`, testApp.Name))
+			Eventually(sess).Should(Exit(0))
 
-		// TODO: does "deis tag" have a k8s implementation?
-		XIt("can set a tag", func() {
+			// find a valid tag to set
+			cmd := "kubectl get nodes -o jsonpath={.items[*].metadata..labels}"
+			// use original $HOME dir or kubectl can't find its config
+			sess, err = start("HOME=%s %s", homeHome, cmd)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Exit(0))
+			// grep output like "map[kubernetes.io/hostname:192.168.64.2 node:worker1]"
+			re := regexp.MustCompile(`([\w\.]{0,253}/?[-_\.\w]{1,63}:[-_\.\w]{1,63})`)
```

Comment:
I think that all cluster nodes should have at least the `kubernetes.io/hostname` label set, which satisfies this test. 

I could move this code into the `BeforeEach` and have the whole spec get `Skip`ped if no labels actually exist, if you think that's better.